### ISSUE TITLE
Compiling for each level increment

### ISSUE THREAD

**Issue by aaapppsss** [nus-cs2113-AY2122S1/forum#17]
Do we have to compile the chatbot using javac each time we push to github?

**Comment by richwill28** [comment]
Compiling with javac produces class files. I'm pretty sure we don't need to push them to GitHub, but I couldn't find <code>*.class</code> on the default <code>.gitignore</code>.

**Comment by nvbinh15** [comment]
Firstly, don't compile the chatbot with javac. Doing so will create `.class` files in the `out/production/ip/` directory, and we don't want to push those files to GitHub. Build and run your program with IntelliJ instead.

After building with IntelliJ, the `.class` file(s) can be found at `out/production/ip/*.class`, and the `/out/` directory is included the `.gitignore` file.

```
# IDEA files
/.idea/
/out/
/*.iml
```

Therefore, the `.class` file(s) will not be pushed to the remote repo, which means that building and running the program before pushing or not does not affect the content you push to GitHub.

TLDR: You don't have to compile the chatbot each time you push.

**Comment by okkhoy** [comment]
In general, it is a good practice to ensure that what you code compiles before pushing it to the remote. 
It becomes imperative in the team project; reason: if you messed up something, it can mess up things for your teammates as well!
In iP it is a bit more relaxed; however, I will encourage you to build (i.e., compile) and test your increment locally.
